vajram/vajram-codegen/src/main/java/com/flipkart/krystal/vajram/codegen/VajramCodeGenerator.java

```diff
@@ -49,140 +65,696 @@
+                                    DEP_RESPONSE, depVariableName));
+                            inputCodeBlocks.add(CodeBlock.builder().add(depVariableName).build());
+                        }
+                    }
+                }
+                returnBuilder.add(codeBlock.build());
+            } else {
+                // call vajram logic method with all input values
+                if (inputDef.isMandatory()) {
+                    inputCodeBlocks.add(CodeBlock.builder().add("element.getInputValueOrThrow($S)", inputDef.name()).build());
+                } else {
+                    inputCodeBlocks.add(CodeBlock.builder().add("element.getInputValueOrDefault($S, null)", inputDef.name()).build());
+                }
+            }
+        });
+        returnBuilder.add("return $L(new $T.AllInputs(\n", vajramDefs.get(vajramName).vajramLogic().getName(),
+                ClassName.get(packageName, CodegenUtils.getInputUtilClassName(vajramName)));
+        // merge the code blocks for inputs
+        for (int i = 0; i<inputCodeBlocks.size(); i++) {
+            // for formatting
+            returnBuilder.add("\t\t");
+            returnBuilder.add(inputCodeBlocks.get(i));
+            if (i != inputCodeBlocks.size()-1) {
+                returnBuilder.add(",\n");
+            }
+        }
+        returnBuilder.add("));\n");
+        returnBuilder.add("}));\n");
+        executeBuilder.addCode(returnBuilder.build());
+
+        return executeBuilder.build();
+    }
+
+    /**
+     * Method to generate "getInputsConvertor" function
+     * @param inputsNeedingModulation Generated Vajram specific InputUtil.InputsNeedingModulation class
+     * @param commonInputs Generated Vajram specific InputUtil.CommonInputs class
+     * @return {@link MethodSpec}
+     */
+    private MethodSpec createInputConvertersMethod(ClassName inputsNeedingModulation, ClassName commonInputs) {
+        Builder inputConvertersBuilder = methodBuilder(METHOD_GET_INPUTS_CONVERTOR).addModifiers(PUBLIC)
+                .returns(ParameterizedTypeName.get(ClassName.get(InputsConverter.class), inputsNeedingModulation, commonInputs))
+                .addAnnotation(Override.class);
+        inputConvertersBuilder.addCode(CodeBlock.builder().addStatement("return $T.CONVERTER", ClassName.get(packageName,
+                CodegenUtils.getInputUtilClassName(vajramName))).build());
+        return inputConvertersBuilder.build();
+    }
+
+    /**
+     * Method to generate "execute" function code for IOVajrams
+     * @param inputsNeedingModulation Generated Vajramspecific InputUtil.InputsNeedingModulation class
+     * @param commonInputs Generated Vajram specific InputUtil.CommonInputs class
+     * @param vajramResponseType Vajram response type
+     * @return generated code for "execute" {@link MethodSpec}
+     */
+    private MethodSpec createIOVajramExecuteMethod(
```

Comment:
Request to change AllInputs to VajramAllInputs and INM to VajramINM and CI to VajramCI.

Reply:
Done